src/Internal/Payment/PaymentContext.php

```diff
@@ -281,6 +281,27 @@ public function get_intent(): ?WC_Payments_API_Abstract_Intention {
 		return $this->get( 'intent' );
 	}
 
+	/**
+	 * Setter for the payment error message.
+	 *
+	 * @param  string $message Error message.
+	 *
+	 * @return void
+	 */
+	public function set_error_message( string $message ): void {
```

Comment:
All the necessary data to determine the context of the error, and why it happened, should already be in the context. Instead of storing errors within it, let's just let the exception bubble up.

Reply:
I am living a bit in the future here and thought that we eventually need to set error message to PaymentContext, similar to what PaymentIntent has - `last_payment_error`
https://github.com/Automattic/woocommerce-payments/blob/ea5370017345fc2254f5011e4fdf381b3842e18a/includes/wc-payment-api/models/class-wc-payments-api-payment-intention.php#L95

But yeah, for now, we do not need it. Just throwing an exception is good enough.

Reply:
We'll probably have something similar, but it would be in the payment processing service, and likely when we start storing the context. Something like:

```php
try {
	return $initial_state->start_processing( $request );
} catch ( Exception $e ) {
	$state = $this->state_factory->createState( $e->is_payment_error() ? PaymentErrorState::class : SystemErrorState::class );
	$state->set_context( $context );

	// Save the state and context.
	$this->storage->save( $context, $state );

	// Log it.
	$this->logger->log( $logs );

	throw $e;
}
```

But until then, errors go straight to the gateway.